apexlog/apexlog.go

```diff
@@ -0,0 +1,96 @@
+package apexlog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/airbrake/gobrake/v5"
+
+	"github.com/apex/log"
+)
+
+// severity identifies the sort of log: info, warn etc.
+type severity int
+
+// These constants identify the log levels in order of increasing severity.
+// A message written to a high-severity log file is also written to each
+// lower-severity log file.
+const (
+	debugLog severity = iota
+	infoLog
+	warnLog
+	errorLog
+	fatalLog
+)
+
+var severityName = []string{
+	debugLog: "debug",
+	infoLog:  "info",
+	warnLog:  "warn",
+	errorLog: "error",
+	fatalLog: "fatal",
+}
+
+// String implementation for severity.
+func (s severity) String() string {
+	return severityName[s]
+}
+
+func parseSeverity(s string) (severity, bool) {
+	for i, name := range severityName {
+		if name == strings.ToLower(s) {
+			return severity(i), true
+		}
+	}
+	return 0, false
+}
+
+var handlerSeverity = errorLog
+
+// SetHandlerSeverity sets minimum log severity that will be sent to Airbrake.
+//
+// Valid names are "debug" "info", "warn", "error", and "fatal".  If the name is not
+// recognized, "error" severity is used.
+func setHandlerSeverity(severity string) {
+	name, ok := parseSeverity(severity)
+	if ok {
+		handlerSeverity = name
+	}
+}
+
+// Handler implementation.
+type Handler struct {
+	Gobrake *gobrake.Notifier
+}
+
+// New Apex Logs handler for airbrake.
+//
+// Valid severity levels are "debug" "info", "warn", "error", and "fatal".  If the name is not
+// recognized, "error" severity is used.
+func New(notifier *gobrake.Notifier, severity string) *Handler {
+	setHandlerSeverity(severity)
+	var v Handler
+	v.Gobrake = notifier
+	return &v
+}
+
+// HandleLog method is used for sending notices to airbrake.
+func (h *Handler) HandleLog(e *log.Entry) error {
+	h.notifyAirbrake(0, severity(e.Level), e.Message)
```

Comment:
We don't need to have `0` (depth) as a function parameter because it never changes. Let's just remove it and hardcode it to 0 on line 93.

Reply:
It's just a variable, so that we can build on it in future if we want to or if we want to add some another variable and need some reference.

Reply:
Could you provide an example of a possible scenario in the future where we want to pass anything other than 0 here?